constraints/src/main/java/gov/nasa/jpl/aerie/constraints/time/IntervalMap.java

```diff
@@ -0,0 +1,437 @@
+      index += 1;
+    }
+
+    // Cases: --[---<---]--->-- and --[---<--->---]--
+    if (index < this.segments.size() && IntervalAlgebra.startsBefore(this.getInterval(index), interval)) {
+      // If the intervals agree on their value, we can unify the old interval with the new one.
+      // Otherwise, we'll snip the old one.
+      if (Objects.equals(this.getValue(index), value)) {
+        interval = IntervalAlgebra.unify(this.segments.remove(index).interval(), interval);
+      } else {
+        final var prefix = IntervalAlgebra.intersect(this.getInterval(index), IntervalAlgebra.lowerBoundsOf(interval));
+        final var suffix = IntervalAlgebra.intersect(this.getInterval(index), IntervalAlgebra.upperBoundsOf(interval));
+
+        this.segments.set(index, Segment.of(prefix, this.getValue(index)));
+        if (!IntervalAlgebra.isEmpty(suffix)) this.segments.add(index + 1, Segment.of(suffix, this.getValue(index)));
+
+        index += 1;
+      }
+    }
+
+    // Cases: --<---[---]--->--
+    while (index < this.segments.size() && !IntervalAlgebra.endsAfter(this.getInterval(index), interval)) {
+      this.segments.remove(index);
+    }
+
+    // Cases: --<---[--->---]--
+    if (index < this.segments.size() && !IntervalAlgebra.startsStrictlyAfter(this.getInterval(index), interval)) {
+      // If the intervals agree on their value, we can unify the old interval with the new one.
+      // Otherwise, we'll snip the old one.
+      if (Objects.equals(this.getValue(index), value)) {
+        interval = IntervalAlgebra.unify(this.segments.remove(index).interval(), interval);
+      } else {
+        final var suffix = IntervalAlgebra.intersect(this.getInterval(index), IntervalAlgebra.upperBoundsOf(interval));
+
+        this.segments.set(index, Segment.of(suffix, this.getValue(index)));
+      }
+    }
+
+    // now, everything left of `index` is strictly left of `interval`,
+    // and everything right of `index` is strictly right of `interval`,
+    // so adding this interval to the list is trivial.
+    this.segments.add(index, Segment.of(interval, value));
+
+    return index;
+  }
+
+  public IntervalMap<V> unset(final Interval interval) {
+    return unset(new IntervalMap<>(interval, new Object()));
+  }
+
+  public IntervalMap<V> unset(final List<Interval> intervals) {
+    return unset(new IntervalMap<>(intervals, new Object()));
+  }
+
+  public IntervalMap<V> unset(final IntervalMap<?> other) {
+    return map2(
+        this, other,
+        (l, r) -> r.isPresent() ? Optional.empty() : l
+    );
+  }
+
+  public IntervalMap<V> select(final Interval bounds) {
+    return unset(List.of(
+        interval(Duration.MIN_VALUE, Inclusive, bounds.start, bounds.startInclusivity.opposite()),
+        interval(bounds.end, bounds.endInclusivity.opposite(), Duration.MAX_VALUE, Inclusive)
+    ));
+  }
+
+  public IntervalMap<V> select(final List<Interval> intervals) {
```

Comment:
I don't see a use case. We select regions of the timeline, which we represent with one or more Intervals. We don't represent regions with IntervalMap.

Reply:
Yeah, I'd argue that either both `select` and `unset` should have `IntervalMap` overloads, or neither of them should, because `unset(IntervalMap)` also doesn't use the actual values of the intervals, just their presence.

I'm okay with either way. :+1:

Reply:
(If you feel like removing them, we can always add them back if we find a use-case. They're pretty simple.)

Reply:
removed